### PR TITLE
multipass: introduce ensure_multipass_is_ready() (CRAFT-438)

### PR DESCRIPTION
**Pull request by cjp256**
Add checks to ensure Multipass is ready for use similar to LXD's.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_providers/multipass/_ready.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import logging
+
+from . import errors
+from .installer import is_installed
+from .multipass import Multipass
+
+logger = logging.getLogger(__name__)
+
+
+def ensure_multipass_is_ready(*, multipass: Multipass = Multipass()) -> None:
```
Comment (sergiusens) [inline craft_providers/multipass/_ready.py:29]: why is this not an instance method in Multipass?

Comment (cjp256) [reply craft_providers/multipass/_ready.py:29]: This was to match the approach used with LXD, it certainly could (should?) be after considering it further.

Comment (cjp256) [reply craft_providers/multipass/_ready.py:29]: This change would result in cross coupling between installer module & Multipass

Comment (sergiusens) [inline craft_providers/multipass/_ready.py:29]: I still wonder why we have a default here and if you could just require it. I would argue that being the of the incorrect version is not not being ready and maybe a name change to `ensure_multipass_environment` or similar would be a bit more appropriate.

**Review by sergiusens — APPROVED** [review]

**Review by facundobatista — APPROVED** [review]
I like it. 

I see the function as "ensure multipass is ready for me to be used" (which includes being a supported version) and not "ready" as in "up and running"...

File: craft_providers/multipass/_ready.py (hunk 1 above)
Comment (cjp256) [reply craft_providers/multipass/_ready.py:29]: I agree the name is a bit odd.  I'd like to wrap up this functionality into a generalized `provider.initialize()` API, but I have been delaying that until things have firmed up on the Charmcraft side and we expand to other craft tools.

The default value is just consistent with the rest of the APIs.  Worth reconsidering that too at some point.

**Pull request merged**